updates to add using statement to vs options docs

dixonsj opened the pull request:
addresses https://github.com/MicrosoftDocs/visualstudio-docs/issues/1614

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/MicrosoftDocs/visualstudio-docs?pullRequest=1668) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/MicrosoftDocs/visualstudio-docs?pullRequest=1668) before we can accept your contribution.<br/><br/>:x: dixonsj [sign now](https://cla.opensource.microsoft.com/MicrosoftDocs/visualstudio-docs?pullRequest=1668)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/MicrosoftDocs/visualstudio-docs?pullRequest=1668) it.</sub>

PRMerger19 commented:
@dixonsj : Thanks for your contribution! The author, @gewarren, has been notified to review your proposed change.

PRMerger8 commented:
Microsoft authors of articles need to make updates through the private repository so the updates can be staged and validated by the current validation rules. After you review staging and fix all validation issues, you can add the sign-off comment to let the PR reviewers know the updates are ready to be merged.It's easy to access the private repo version of your articles in GitHub. Just add -pr to the URL, as follows: 
 Public repo URL: https://github.com/Microsoft/azure-docs/blob/master/articles/azure-portal-overview.md 
 Private repo URL: https://github.com/Microsoft/azure-docs-pr/blob/master/articles/azure-portal-overview.md 
 We have closed this pull request. Please resubmit the updates through the private repo. If you need perms to the private repo, [see the contributor guide instructions](https://github.com/Microsoft/azure-docs/blob/master/contributor-guide/tools-and-setup.md#permissions)

gewarren commented on docs/ide/reference/options-text-editor-csharp-advanced.md:
````diff
@@ -26,6 +26,70 @@ Use the **Advanced** options page to modify the settings for editor formatting,
+   using System.Linq;
+   ```
+   
+- Add usings for types in reference assemblies and NuGet packages 
````
From the name of this option, I'm not sure that the description that VS installs the NuGet package is correct. I think it's more likely that this option means that VS will suggest using directives to add for types you're using that are in referenced assemblies or NuGet packages - not that it will install the NuGet package.

dixonsj replied:
Hi, that's exactly what it does. 

If you have the option selected it will look for and install types from un-referenced and installed nuget packages.

I created a new blank console app and if add the code var context = new DbContext, the lightbulb will allow me to install the latest version of the nuget package, or use the package manager to install a specific version.

as you can see in the screenshot below, i dont have any nuget pkgs installed but it is now allowing me install the EntityFramework pkg.

![image](https://user-images.githubusercontent.com/16678850/46836700-5a968000-cdfe-11e8-9356-a8c21736d177.png)

kendrahavens replied:
@dixonsj is correct! We added the option in 2015.3 ([release notes](https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2015-update3-vs#csharpandvb)) though it's never been turned on by default! Very useful and we definitely need to document this better. The only other place it's mentioned that I could find is one of Scott Hansleman's[ blog posts.](https://www.hanselman.com/blog/VisualStudio2017CanAutomaticallyRecommendNuGetPackagesForUnknownTypes.aspx)

kendrahavens commented on docs/ide/reference/options-text-editor-csharp-advanced.md:
```diff
@@ -30,56 +30,64 @@ Use the **Advanced** options page to modify the settings for editor formatting,
-    using FluentValidation;
-    using Newtonsoft.Json;
-
+   When selected, when you select **Remove and Sort Usings**, Visual Studio sorts the `using` directives and places the 'System' namespaces at the top of the list.
```
Could we change the phrase "When selected, when you select..." to "When selected, the **Remove and Sort Usings** option in the right-click menu sorts the 'using' directives and place the `System` namespaces at the top of the list"

kendrahavens commented on docs/ide/reference/options-text-editor-csharp-advanced.md:
```diff
@@ -30,56 +30,64 @@ Use the **Advanced** options page to modify the settings for editor formatting,
-    using System.Collections.Generic;
-    using System.Linq;
-
+   When selected, when you select **Remove and Sort Usings**, Visual Studio separates `using` directives by inserting an empty line between groups of directives.
```
Same here.

Reword the phrase "When selected, when you select..." to "When selected, the **Remove and Sort Usings** option in the right-click menu separates 'using' directives by inserting an empty line between groups of directives."

PRMerger10 commented:
@dixonsj : Thanks for your contribution! The author, @gewarren, has been notified to review your proposed change.

dixonsj commented:
Hi,

I have updated as requested.

Thanks,

Shane.

From: Kendra Havens <notifications@github.com>
Sent: Friday, 12 October 2018 11:28 AM
To: MicrosoftDocs/visualstudio-docs <visualstudio-docs@noreply.github.com>
Cc: Shane Dixon <shane.dixon@outlook.com>; Mention <mention@noreply.github.com>
Subject: Re: [MicrosoftDocs/visualstudio-docs] updates to add using statement to vs options docs (#1668)


@kendrahavens commented on this pull request.

Everything looks good, I left a few comments on wording. Thank you!

@jinujoseph<https://github.com/jinujoseph> and myself can be included on productivity-related docs.

________________________________

In docs/ide/reference/options-text-editor-csharp-advanced.md<https://github.com/MicrosoftDocs/visualstudio-docs/pull/1668#discussion_r224640429>:

> -    using FluentValidation;

-    using System.Collections.Generic;

-    using System.Linq;

-    using Newtonsoft.Json;

-    using System;

-

-After Sort

-

-    using System;

-    using System.Collections.Generic;

-    using System.Linq;

-    using AutoMapper;

-    using FluentValidation;

-    using Newtonsoft.Json;

-

+   When selected, when you select **Remove and Sort Usings**, Visual Studio sorts the `using` directives and places the 'System' namespaces at the top of the list.

Could we change the phrase "When selected, when you select..." to "When selected, the Remove and Sort Usings option in the right-click menu sorts the 'using' directives and place the System namespaces at the top of the list"

________________________________

In docs/ide/reference/options-text-editor-csharp-advanced.md<https://github.com/MicrosoftDocs/visualstudio-docs/pull/1668#discussion_r224640608>:

> -    using Newtonsoft.Json;

-    using System;

-

-After Sort

-

-    using AutoMapper;

-

-    using FluentValidation;

-

-    using Newtonsoft.Json;

-

-    using System;

-    using System.Collections.Generic;

-    using System.Linq;

-

+   When selected, when you select **Remove and Sort Usings**, Visual Studio separates `using` directives by inserting an empty line between groups of directives.

Same here.

Reword the phrase "When selected, when you select..." to "When selected, the Remove and Sort Usings option in the right-click menu separates 'using' directives by inserting an empty line between groups of directives."

________________________________

In docs/ide/reference/options-text-editor-csharp-advanced.md<https://github.com/MicrosoftDocs/visualstudio-docs/pull/1668#discussion_r224641883>:

> +

+   After sorting:

+

+   ```csharp

+   using AutoMapper;

+

+   using FluentValidation;

+

+   using Newtonsoft.Json;

+

+   using System;

+   using System.Collections.Generic;

+   using System.Linq;

+   ```

+

+- Add usings for types in reference assemblies and NuGet packages

@dixonsj<https://github.com/dixonsj> is correct! We added the option in 2015.3 (release notes<https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2015-update3-vs#csharpandvb>) though it's never been turned on by default! Very useful and we definitely need to document this better. The only other place it's mentioned that I could find is one of Scott Hansleman's blog posts.<https://www.hanselman.com/blog/VisualStudio2017CanAutomaticallyRecommendNuGetPackagesForUnknownTypes.aspx>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/MicrosoftDocs/visualstudio-docs/pull/1668#pullrequestreview-164060700>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AP5_wsXUq4fq2FnSoe7kczKWXpC-FLzqks5uj-IogaJpZM4XUM0q>.

PRMerger7 commented:
Microsoft authors of articles need to make updates through the private repository so the updates can be staged and validated by the current validation rules. After you review staging and fix all validation issues, you can add the sign-off comment to let the PR reviewers know the updates are ready to be merged.It's easy to access the private repo version of your articles in GitHub. Just add -pr to the URL, as follows: 
 Public repo URL: https://github.com/Microsoft/azure-docs/blob/master/articles/azure-portal-overview.md 
 Private repo URL: https://github.com/Microsoft/azure-docs-pr/blob/master/articles/azure-portal-overview.md 
 We have closed this pull request. Please resubmit the updates through the private repo. If you need perms to the private repo, [see the contributor guide instructions](https://github.com/Microsoft/azure-docs/blob/master/contributor-guide/tools-and-setup.md#permissions)

PRMerger10 commented:
Microsoft authors of articles need to make updates through the private repository so the updates can be staged and validated by the current validation rules. After you review staging and fix all validation issues, you can add the sign-off comment to let the PR reviewers know the updates are ready to be merged.It's easy to access the private repo version of your articles in GitHub. Just add -pr to the URL, as follows: 
 Public repo URL: https://github.com/Microsoft/azure-docs/blob/master/articles/azure-portal-overview.md 
 Private repo URL: https://github.com/Microsoft/azure-docs-pr/blob/master/articles/azure-portal-overview.md 
 We have closed this pull request. Please resubmit the updates through the private repo. If you need perms to the private repo, [see the contributor guide instructions](https://github.com/Microsoft/azure-docs/blob/master/contributor-guide/tools-and-setup.md#permissions)

gewarren commented:
@dixonsj Thanks for adding these options!

gewarren commented:
@kendrahavens Thanks for reviewing, that was awesome!

dixonsj commented:
Um. The image doesnt work on master?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
From: Genevieve Warren <notifications@github.com>
Sent: Tuesday, October 16, 2018 5:19:34 AM
To: MicrosoftDocs/visualstudio-docs
Cc: Shane Dixon; Mention
Subject: Re: [MicrosoftDocs/visualstudio-docs] updates to add using statement to vs options docs (#1668)


@kendrahavens<https://github.com/kendrahavens> Thanks for reviewing, that was awesome!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/MicrosoftDocs/visualstudio-docs/pull/1668#issuecomment-429960458>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AP5_wlTtOUNvDgOZabKR6Gf7lRXILkg9ks5ulNG2gaJpZM4XUM0q>.

gewarren commented:
Thanks for noticing that. It's fixed now.
